public/static/app/common/EvgUiGridUtil.js

```diff
@@ -0,0 +1,63 @@
+mciModule.factory('EvgUiGridUtil', function() {
+  // Allows to get ui grid col options
+  // :parm gridOptions: ui grid options
+  // :returns: Function (with gridOptions stored in the context)
+  function getColAccessor(gridOptions) {
+    // :param fieldName: ui grid field name
+    // :returns: ui grid col object
+    return function(fieldName) {
+      return _.findWhere(gridOptions.columnDefs, {
+        field: fieldName
+      })
+    }
+  }
+
+  // Applies filter options to ui grid columns
+  // Options is unique set of unique items for each given field
+  // :param field: List of strings (field names)
+  // :param gridOptions: ui grid options object
+  // :returns: none
+  function applyMultiselectOptions(data, fields, gridOptions) {
+    // Create a column accessor function
+    var colAccessor = getColAccessor(gridOptions)
+
+    // Apply options data to column options
+    _.each(fields, function(d) {
+      colAccessor(d).filter.options = _.unique(_.pluck(data, d))
+    })
+  }
+
+  // Does row filtering for ui grid. List of params is compatible
+  // with ui-grid condition function format
+  // :param term: array of selected 'terms'
+  // :returns: True if val is equal to at least one of the terms.
+  //           False otherwise
+  function multiselectConditionFn(term, val, row, col) {
+    if (typeof(term) == 'undefined') return true
```

Comment:
I would do if (!term || term.length === 0)

Reply:
@john-m-liu , works fine on FF and Chrome. But I noticed another little issue with search. BTW, using `typeof(term) == 'undefined'` is only reliable way to check if the object is not defined in JS. However, in most cases, including my case, your option will also work fine.
I will be happy if we start use coffee script...